src/gt4py/next/program_processors/runners/dace_iterator/itir_to_tasklet.py

```diff
@@ -350,6 +351,104 @@ def builtin_undefined(*args: Any) -> Any:
 }
 
 
+class GatherLambdaSymbolsPass(eve.NodeVisitor):
+    sdfg: dace.SDFG
+    state: dace.SDFGState
+    symbol_map: dict[str, IteratorExpr | SymbolExpr | ValueExpr]
+    parent_symbol_map: dict[str, IteratorExpr | SymbolExpr | ValueExpr]
+
+    def __init__(
+        self,
+        sdfg,
+        state,
+        symbol_map,
```

Comment:
You pass `symbol_map` as a `&` to the constructor and in the code after calling `visit` you rely on the fact that `GatherLambdaSymbolsPass` will modify the original `symbol_map`. If `GatherLambdaSymbolsPass` made a (shallow) copy of your `symbol_map`, your code would break despite no API changes as the user of the *pass* would not see the changes to the dict. I think it'd better to remove this parameter from the ctor, create an empty dict inside the ctor, and retrieve the filled dict through a property after calling `visit`.

Reply:
Right, it makes sense